src/tensors/planar.jl

```diff
@@ -322,3 +322,67 @@ function planar_contract!(α, A::AbstractTensorMap{S}, B::AbstractTensorMap{S},
     # end
     return C
 end
+
+function planar_trace!(α, tsrc::AbstractTensorMap{S}, CA::Symbol,
+                       β, tdst::AbstractTensorMap{S, N₁, N₂},
+                       p1::IndexTuple{N₁}, p2::IndexTuple{N₂},
+                       q1::IndexTuple{N₃}, q2::IndexTuple{N₃}) where {S, N₁, N₂, N₃}
+    if BraidingStyle(sectortype(S)) == Bosonic()
+        return TO.trace!(α, tsrc, CA, β, tdst, p1, p2, q1, q2)
+    end
+
+    if CA == :N
+        p = (p1..., p2...)
+        pl = TupleTools.getindices(p, codomainind(tdst))
+        pr = TupleTools.getindices(p, domainind(tdst))
+        _planar_trace!(α, tsrc, β, tdst, pl, pr, q1, q2)
+    else
+        p = adjointtensorindices(tsrc, (p1..., p2...))
+        pl = TupleTools.getindices(p, codomainind(tdst))
+        pr = TupleTools.getindices(p, domainind(tdst))
+        q1 = adjointtensorindices(tsrc, q1)
+        q2 = adjointtensorindices(tsrc, q2)
+        _planar_trace!(α, adjoint(tsrc), β, tdst, pl, pr, q1, q2)
+    end
+end
```

Comment:
Most of this can go, and `_planar_trace!` can just become `planar_trace!`. Come to think about it, the macro preprocessing does however not check whether traces are compatible with planarity, so this I will need to add.